iconrpcserver/server/rest_server.py

```diff
@@ -155,7 +155,8 @@ class Avail(HTTPMethodView):
         status = HTTPStatus.OK
         result = PeerServiceStub().get_status(channel_name)
 
-        if result['status'] != "Service is online: 0":
+        available_states = ["BlockGenerate", "Vote", "LeaderComplain", "Watch"]
```

Comment:
Where is source of available states?
Is it possible to get available state from defined source?
If not, I recommend check available in source of state defined and return available or not.

Reply:
Code updated.

- Now Avail api responses status code in case of 'is_available' value is True
- The channel state machine in loopchain decides which states are available.